src/sdk/pynni/nni/hyperopt_tuner/hyperopt_tuner.py

```diff
@@ -220,32 +220,17 @@ def generate_parameters(self, parameter_id):
         -------
         params : dict
         """
-        rval = self.rval
-        trials = rval.trials
-        algorithm = rval.algo
-        new_ids = rval.trials.new_trial_ids(1)
-        rval.trials.refresh()
-        random_state = rval.rstate.randint(2**31-1)
-        new_trials = algorithm(new_ids, rval.domain, trials, random_state)
-        rval.trials.refresh()
-        vals = new_trials[0]['misc']['vals']
-        parameter = dict()
-        for key in vals:
-            try:
-                parameter[key] = vals[key][0].item()
-            except Exception:
-                parameter[key] = None
-
-        # remove '_index' from json2parameter and save params-id
-        total_params = json2parameter(self.json, parameter)
+        total_params = self.get_suggestion(random_search=False)
```

Comment:
please add comments here to tell why making this change, and that it is still possible to generate duplicate configurations.

Reply:
Ok